### PR TITLE
Support for stopping certain git users from triggering builds (ie, the Jenkins user)

### PR DESCRIPTION
**Pull request by dbsanfte**
This PR provides support for stopping certain git users from triggering builds on a push event.

### CODE REVIEW

**Comment by tobilarscheid** [comment]
Nice! Thank you!

**Comment by dbsanfte** [comment]
Had to make a change, I noticed tag create/delete events by the ignored users were still triggering builds.

**Comment by stephenc** [comment]
This PR will not be merged as it currently stands. This PR is basically hacking in changes rather than abstracting the changes into an API that can be more generally used by others. https://github.com/jenkinsci/github-branch-source-plugin/pull/162 provides an example of how changes can be made.

**Comment by AnneTheAgile** [comment]
@stephenc where is the  the BranchBuildStrategy api  code located?

**Comment by b-b3rn4rd** [comment]
@stephenc in order to run a `changesSince` I need to know a previous revision as well, otherwise, it returns all commits made from the beginning.
```
            ByteArrayOutputStream out = new ByteArrayOutputStream();
            // how to get prev revision id ?
            Boolean hasChanges = fileSystem.changesSince(null, out);            
```
Could you please explain how I can get a previous revision from the current one

**Comment by stephenc** [comment]
@b-b3rn4rd when https://github.com/jenkinsci/branch-api-plugin/pull/119/files is merged you will be able to do that

**Comment by stephenc** [comment]
@AnneTheAgile https://github.com/jenkinsci/branch-api-plugin/blob/master/src/main/java/jenkins/branch/BranchBuildStrategy.java

**Comment by b-b3rn4rd** [comment]
That's exactly what I needed!  Another way I was thinking is to  retrieve a `job -> getLastSuccessfulBuild -> getChangeSet `, but my concern was that changeset is a Build specific thing where is we are dealing with SCM

**Comment by stephenc** [comment]
@b-b3rn4rd released as branch-api 2.0.17

**Comment by jglick** [comment]
The UI here implies that restriction is done by Git committer, which is not correct. It is done by the user pushing commits, which may be totally unrelated.

Suppressing builds due to Git committer, which is what people usually actually want, is a longstanding RFE with discussion in [JENKINS-35988](https://issues.jenkins-ci.org/browse/JENKINS-35988).

**Comment by stephenc** [comment]
And git committer really needs GPG signed commits otherwise it can be faked (more of an issue for an inverse filter)

**Comment by b-b3rn4rd** [comment]
Using the suggested approach I developed https://github.com/jenkinsci/ignore-committer-strategy-plugin, which prevents multibranch projects from triggering new builds based on a blacklist of commit authors in a changeset, however, it's not exactly what the author of this PR is trying to achieve...
In our scenario, we just wanted to prevent new builds when Jenkins commits/pushes during artifact build process.

**Comment by quike** [comment]
Hi! any update on this?

**Comment by pinturic** [comment]
Hi @b-b3rn4rd  can you please detail the difference between your approach and this? However your plugins looks really promising

File: src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration/help-usersToIgnore.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    <p>
+        A comma-delimited list of Git usernames to ignore push events from. These events will be handled
+        and logged by the plugin but will not trigger builds. The obvious use-case for this is
+        ignoring push events by a designated Jenkins git user, so that your builds don't trigger
+        more builds.
+    </p>
+    <p>
+        This must <strong>exactly</strong> match their Github username.
+    </p>
+    <p>
```
Comment (recena) [inline src/main/resources/org/jenkinsci/plugins/github_branch_source/GitHubConfiguration/help-usersToIgnore.html]: I recommend you to use:
```
<dl>
  <dt>Examples:</dt>
  <dd>jenkinsuser</dd>
  <dd>jenkinsuser,otheruser,anotheruser,yetanotheruser</dd>
</dl>
```

The proposed HTML is wrong from a semantic perspective.

**Comment by bitwiseman** [comment]
@stephenc @dbsanfte @recena 
It looks like this is stale should we close?

**Comment by stephenc** [comment]
I agree on closing

**Pull request closed without merging**